### DISCUSSION TITLE
2.18.1-2

### DISCUSSION THREAD

**Discussion by bb-qq** [bb-qq/r8152#441] [Announcements]
* Fixed the MTU issue for RTL8157. (Thanks to @wisdpi)
* Added support of Alienware AW1022 2.5GbE.

Note:

* The installation will fail the first time. After that, run the following command from the SSH terminal:
   `sudo install -m 4755 -o root -D /var/packages/r8152/target/r8152/spk_su /opt/sbin/spk_su`
* Selecting a proper version is needed depending on your DSM version.
  * DSM 7.2 and above
    * use packages with the suffix `_7.2`.
  * DSM 7.0 and DSM 7.1
    * use packages with the suffix `_7.1`.
  * DSM 6.2.x
    * use packages with the suffix `_6.2`.

<hr /><em>This discussion was created from the release <a href='https://github.com/bb-qq/r8152/releases/tag/'>2.18.1-2</a>.</em>

**Comment by TheS1R** [comment]
I updated driver to 2.18.1-2 on my DS 718+ (newest DSM 7.2.2-72806). The DSM GUI still does not show option to select MTU higher than 1500.

**Reply by TheS1R** [reply]
**UPDATE**: MTU can be set to jumbo frame via command line (_verify device name, and substitute as necessary_):

`sudo ifconfig eth2 mtu 9000`

**Comment by dffvb** [comment]
I dont care about this release, just wanted to say big thank you, whenever I upload data and get these 270 MB/s I praise you. Thank you.

**Comment by Roendi61** [comment]
HI
I have a Sysnology DS1019 with a Realtek Gaming USB 2.5GbE Family Controller Chip inside from Wavlink. My old Switch support only 2.5 GB and I see all Information about the Connection. Today my new 10 GB Switch is coming. When I us a 10 Gig Port the Network Status is not showing

When I use a 2.5 Gig Port all is fine. I can change the MTU what I want and all Settings will showing korrekt

I tried first all 7.2 DSM Driver to install. For me is was only Abel this driver to install (r8152-apollolake-2.18.1-2_7.2)
ON all other 7.2 Driver Synology say. Is not for this system.


When I connect the NAS on the 2.5 GB Port i see this

Settings for eth2:
        Supported ports: [ MII ]
        Supported link modes:   10baseT/Half 10baseT/Full
                                100baseT/Half 100baseT/Full
                                1000baseT/Full
                                2500baseX/Full
        Supported pause frame use: No
        Supports auto-negotiation: Yes
        Supported FEC modes: Not reported
        Advertised link modes:  10baseT/Half 10baseT/Full
                                100baseT/Half 100baseT/Full
                                1000baseT/Full
                                2500baseX/Full
        Advertised pause frame use: Symmetric Receive-only
        Advertised auto-negotiation: Yes
        Advertised FEC modes: Not reported
        Link partner advertised link modes:  10baseT/Half 10baseT/Full
                                             100baseT/Half 100baseT/Full
                                             1000baseT/Full
        Link partner advertised pause frame use: No
        Link partner advertised auto-negotiation: Yes
        Link partner advertised FEC modes: Not reported
        Speed: 2500Mb/s
        Duplex: Full
        Port: MII
        PHYAD: 32
        Transceiver: internal
        Auto-negotiation: on
        Supports Wake-on: pumbg
        Wake-on: d
        Current message level: 0x00007fff (32767)
                               drv probe link timer ifdown ifup rx_err tx_err tx                                                        _queued intr tx_done rx_status pktdata hw wol
        Link detected: yes

This will not working
ethtool -s eth2 speed 5000 duplex full autoneg off
Cannot set new settings: Invalid argument
  not setting speed
  not setting duplex
  not setting autoneg


Same this
ethtool -s  eth2 autoneg on advertise 0x180000000002f
Cannot set new settings: Operation not supported
not setting autoneg

Thanks for your great work

**Comment by darklordvader77** [comment]
Installed [r8152-r1000-2.18.1-2_7.2.spk] on my DS1522+ and everything seems to work on a driver level. The package is auto-running after reboot or shutdown-startup state correctly. I'm waiting for a 2.5GB Switch later this week to begin testing. 

The adapter is a UGREEN 2.5GB to USB-A (RTL8156BG) and i bought it here: https://www.amazon.de/-/en/dp/B0CWV2Q6HJ

Thanks bb-qq for your great work!

**Comment by A380kadse** [comment]
Tested for the first time on a ds423+ with a Club USB 3.2 Gen1 Type -A 2.5Gbps and it works stable. 
Many thanks!

**Comment by Flavien55** [comment]
hello i ve a synology ds423+ with a asus usb-c2500 

my DSM 7.2.2-72806

i download
r8152-alpine-2.18.1-2_7.2.spk

when i try to install i get this message 

This package is not supported on the Synology NAS platform or is incompatible with the current version of DSM


what i can do ?

**Reply by A380kadse** [reply]
You are using the wrong spk
the ds423+ is based on geminilake

<img width="452" alt="Bildschirmfoto 2024-11-04 um 12 21 35" src="https://github.com/user-attachments/assets/2f1d99a9-40b5-4ff9-ad0e-617ff27dcf53">
<img width="1154" alt="Bildschirmfoto 2024-11-04 um 12 24 19" src="https://github.com/user-attachments/assets/89fbef8c-c857-4247-b195-043855fff873">

**Comment by arhoj** [comment]
No issues here – works a charm as always. Just wanted to thank you for your work 🙌

**Comment by Moxie-91** [comment]
Hello dear people I need help, I cant find my procesor on the list of driver or that procesor is not supported at all .
I have Synology DS413 and on Synology sight  it say that DS413 have  Freescale P1022 Qoriq dual core.
Last DSM version for this model is 6.2.4 
I have Asus ASUS USB-C2500 USB 2.5Gb card that I would like to plug in this NAS.

**Reply by bb-qq** [reply]
Unfortunately, the kernel version of this platform is too old for the driver to be supported.

**Reply by Moxie-91** [reply]
Well thanks either way, you've made a lot of people happy, unfortunately I won't be one of them (I understand and thanks for answering my question), still thanks for your work and effort to bring some life back to the platform.

**Comment by jokinblanco** [comment]
Hello I made big mistake and run the command before the first failed instalation. Looks like now I cant install it properly... Any solutions please. Big thanks

**Reply by bb-qq** [reply]
Even if the installation has failed, the `install`  command itself executed via SSH should be able to complete successfully. Can you describe in detail what is not working?

**Reply by jokinblanco** [reply]
Thanks for your answer! It is working now. With iperf i get 2350 mb/s on upload and 1940 on download . Do you know what can be the reason. Thanks for your work!!
